app/src/main/java/org/sil/gatherwords/SessionActivity.java

```diff
@@ -206,4 +232,42 @@ public void onItemSelected(AdapterView<?> parent, View view,
     public void onNothingSelected(AdapterView<?> parent) {
         worldListToLoad = "";
     }
+
+
+    //TODO: Rename
+    private static class GetInfoFromDB extends AsyncTask<Integer, Void, List<Session>> {
+        private SessionDao sDAO;
+        private WeakReference<SessionActivity> sessionActivityRef;
+
+        GetInfoFromDB(SessionActivity sessionActivity) {
+            sDAO = AppDatabase.get(sessionActivity).sessionDao();
+            sessionActivityRef = new WeakReference<>(sessionActivity);
+        }
+
+        @Override
+        protected List<Session> doInBackground(Integer... ids) {
+            return sDAO.getSessionsByID(ids);
+        }
+
+        @Override
+        protected void onPostExecute(List<Session> sessions) {
+            SessionActivity sessionActivity = sessionActivityRef.get();
+            if (sessionActivity != null && sessions != null && sessions.size() == 1 ) {
+                Session session = sessions.get(0);
+                // Insert previous date
+                Date date = session.date;
+                sessionActivity.dateField.setText(sessionActivity.dateSDF.format(date));
+                sessionActivity.timeField.setText(sessionActivity.timeSDF.format(date));
+                String timeZoneString = sessionActivity.timeZoneSDF.format(date);
+                sessionActivity.timeZoneField.setText(timeZoneString.substring(3, timeZoneString.length()));
+
+                // Insert previous lable, speaker, eliciter
+                sessionActivity.labelField.setText(session.label);
+                sessionActivity.speakerField.setText(session.speaker);
+                sessionActivity.eliciterField.setText(session.recorder);
+            } else {
+                Log.e("SessionActivity", "empty or size>1 Session[] grabbed from database");
```

Comment:
We should not report this if `sessionActivty` is `null`, since that simply means the user switched pages/closed the app while the db was running.